### PR TITLE
Render AMIs into CloudFormation template after copy

### PR DESCRIPTION
**Pull request by ghost**
This PR ensures that `conjur.yml` is rendered with all AMIs, if we have chosen to copy from us-east-1 to other regions. This happens by re-rendering the cloudformation template after copying the AMIs to other regions.

The big user-facing change is that users no longer have to enter the image ID, it will be autodetected from their AWS region. To that end, we should start versioning `conjur.yml`, the output template, in line with cyberark/conjur versioning. That will make this easier to track and distribute.

### CODE REVIEW

**Comment by dustinmm80** [comment]
Build is green in Jenkins, this is ready for review.

https://jenkins.conjur.net/blue/organizations/jenkins/cyberark--conjur-aws/detail/render-cloudformation/13/pipeline

**Comment by dustinmm80** [comment]
You can see the generated CFN template in Jenkins: https://jenkins.conjur.net/job/cyberark--conjur-aws/job/render-cloudformation/13/artifact/conjur.yml

File: promote-to-regions.sh [hunk 1]
```diff
@@ -22,8 +22,14 @@ docker pull $PROMOTER_IMAGE
...
+  | tee amis.json
 
-# Add the source AMI to the list so it is complete
-source_ami_string="{\"us-east-1\": \"$SOURCE_AMI\","
-sed "1s/^./$source_ami_string /" AMIS.json
+# Convert ami-promoter output to YML, for easier merging
+docker run --rm \
+  -v $PWD:/src -w /src \
+  ruby:2.3 \
+  ruby -ryaml -rjson -e 'puts YAML.dump(JSON.parse(STDIN.read))' < amis.json > vars-amis.yml
```
Comment (dividedmind) [inline promote-to-regions.sh]: Great trick, I love it! YAML FTW

**Review by dividedmind — CHANGES_REQUESTED** [review]
Looks good to me, thanks!
One change I'd like to see -- since you mention versioning the CF template with Conjur. Correct me if I'm wrong, but I believe this job already knows Conjur version it's building. Can it use that knowledge to rename the output template to `conjur-$version.yml`?

**Comment by dustinmm80** [comment]
We are now writing `conjur-$version.yml` cloudformation template files, along with `conjur.yml`. `conjur.yml` now acts as a local artifact, it is not published to S3. I think this is ready to merge.

https://jenkins.conjur.net/job/cyberark--conjur-aws/job/render-cloudformation/15/

**Comment by dividedmind** [comment]
Merged, thanks!

**Pull request merged**